src/test/java/it/gov/pagopa/payment/service/QRCodePaymentServiceImplTest.java

```diff
@@ -0,0 +1,102 @@
+package it.gov.pagopa.payment.service;
+
+import it.gov.pagopa.payment.dto.mapper.TransactionInProgress2SyncTrxStatusMapper;
+import it.gov.pagopa.payment.dto.mapper.TransactionInProgress2SyncTrxStatusTest;
+import it.gov.pagopa.payment.dto.qrcode.SyncTrxStatusDTO;
+import it.gov.pagopa.payment.enums.SyncTrxStatus;
+import it.gov.pagopa.payment.exception.ClientExceptionNoBody;
+import it.gov.pagopa.payment.model.TransactionInProgress;
+import it.gov.pagopa.payment.repository.TransactionInProgressRepository;
+import it.gov.pagopa.payment.service.qrcode.QRCodeAuthPaymentService;
+import it.gov.pagopa.payment.service.qrcode.QRCodeConfirmationService;
+import it.gov.pagopa.payment.service.qrcode.QRCodeCreationService;
+import it.gov.pagopa.payment.service.qrcode.QRCodePreAuthService;
+import it.gov.pagopa.payment.test.fakers.TransactionInProgressFaker;
+import it.gov.pagopa.payment.utils.RewardConstants;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.HttpStatus;
+
+import java.time.LocalDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class QRCodePaymentServiceImplTest {
+    @Mock
+    TransactionInProgressRepository transactionInProgressRepositoryMock;
+    @Mock
+    QRCodeCreationService qrCodeCreationServiceMock;
+    @Mock
+     QRCodePreAuthService qrCodePreAuthServiceMock;
+    @Mock
+    QRCodeAuthPaymentService qrCodeAuthPaymentServiceMock;
+    @Mock
+    QRCodeConfirmationService qrCodeConfirmationServiceMock;
+    @Spy
+    private final TransactionInProgress2SyncTrxStatusMapper transactionMapper= new TransactionInProgress2SyncTrxStatusMapper();
+
+    QRCodePaymentServiceImpl qrCodePaymentServiceMock;
+
+    @BeforeEach
+    void setUp(){
+        qrCodePaymentServiceMock = new QRCodePaymentServiceImpl(qrCodeCreationServiceMock,
+                qrCodePreAuthServiceMock,
+                qrCodeAuthPaymentServiceMock,
+                qrCodeConfirmationServiceMock,
+                transactionInProgressRepositoryMock,
+                transactionMapper);
+    }
+
+    @AfterEach
+    void tearDown() {
```

Comment:
why this name?
```suggestion
    void verifyNoMoreMockInteractions() {
```

Reply:
the name of the tearDown method is the provided by junit, it is the same thing as the setUp

Reply:
ok, let's rename it in order to give an hint on its purpose